### ISSUE TITLE
Forward+ render projects not opening in AMD laptop

### ISSUE THREAD

**Issue by ahmed-mo2nis** [godotengine/godot#87909]
### Tested versions

Godot 4.2.1 stable

### System information

Godot v4.2.1.stable - Pop!_OS 22.04 LTS - X11  - RENOIR (radeonsi, renoir, LLVM 15.0.7, DRM 3.54, 6.6.10-76060610-generic) () - AMD Ryzen 9 5900HS with Radeon Graphics (16 Threads)

### Issue description

When I create a new project with Forward+ renderer or change the renderer of a project from Compatibility to Forward+ I get this blank screen and the engine freezes whenever I try to open the project.
![Screenshot from 2024-02-03 19-51-44](https://github.com/godotengine/godot/assets/40705255/687e4938-7ed5-41bb-a5e2-4922e3226827)

### Steps to reproduce

1-Create a new project in an AMD Ryzen laptop
2-Choose a Forward+ renderer
3-Open the project

### Minimal reproduction project (MRP)

Just an empty project since it doesn't open.

**Comment by ryevdokimov** [comment]
Unable to reproduce in Windows, must be something with how your OS is setup. 

I have a Thinkpad T14s AMD that also only has an iGPU. 

Godot v4.2.1.stable.mono - Windows 10.0.22631 - Vulkan (Forward+) - integrated AMD Radeon(TM) Graphics (Advanced Micro Devices, Inc.; 31.0.23013.1023) - AMD Ryzen 7 PRO 6850U with Radeon Graphics (16 Threads)

**Comment by Calinou** [comment]
Can you reproduce this when creating a project with the Mobile rendering method?

**Comment by gregkrsak** [comment]
I have this same issue on Windows 10 Pro, using an Alienware Alpha (GeForce GTX 860m) with 16GB of RAM. Switching to Mobile fixes it. Can't start Godot (either from Steam or from an individual build) without a command line or project setting set to change the renderer.

**Comment by ahmed-mo2nis** [comment]
> Can you reproduce this when creating a project with the Mobile rendering method?

I tried to create a new project. Same problem with Mobile renderer.

**Comment by Calinou** [comment]
> Can't start Godot (either from Steam or from an individual build) without a command line or project setting set to change the renderer.

The project manager always uses the Compatibility rendering method, unless you use `--rendering-driver vulkan` to force it to use Vulkan (in which case it'll use Forward+ unless `--rendering-method mobile` is specified).

**Comment by gregkrsak** [comment]
I'm pretty sure that's backwards. I have to add the option for
compatibility mode to the Steam launch options or Godot 4.2 crashes on
launch in Windows.

**Comment by ahmed-mo2nis** [comment]
I attempted something different today. I downloaded the TPS template from [here](https://godotengine.org/asset-library/asset/678). I opened the project and it worked even though it's a Forward+ renderer so that means the problem is only when starting a new project or changing the renderer of an existing project.

**Comment by clayjohn** [comment]
@ahmed-mo2nis Can you run Godot from the command line with the `--verbose` option and then let us know what GPUs are detected by the engine. From your original post, it looks like the engine isn't running on your GPU, which may be the source of your issue. 

The full command is:
```
./Godot_v4.2.1-stable_linux.x86_64 --verbose
```
It will print out a lot of information. After you open the TPS demo, if you scroll up you should find a section that looks like:
```
Vulkan devices:
  #0: Intel Intel(R) Xe Graphics (TGL GT2) - Supported, Integrated
Vulkan API 1.3.267 - Forward Mobile - Using Vulkan Device #0: Intel - Intel(R) Xe Graphics (TGL GT2)
```
(This is how it appears for me with a project using the Mobile renderer)

**Comment by ahmed-mo2nis** [comment]
> @ahmed-mo2nis Can you run Godot from the command line with the `--verbose` option and then let us know what GPUs are detected by the engine. From your original post, it looks like the engine isn't running on your GPU, which may be the source of your issue.
> 
> The full command is:
> 
> ```
> ./Godot_v4.2.1-stable_linux.x86_64 --verbose
> ```
> 
> It will print out a lot of information. After you open the TPS demo, if you scroll up you should find a section that looks like:
> 
> ```
> Vulkan devices:
>   #0: Intel Intel(R) Xe Graphics (TGL GT2) - Supported, Integrated
> Vulkan API 1.3.267 - Forward Mobile - Using Vulkan Device #0: Intel - Intel(R) Xe Graphics (TGL GT2)
> ```
> 
> (This is how it appears for me with a project using the Mobile renderer)
> 
> Could you also try doing the same thing, but opening a project that fails to load?

I tried creating a new project after opening Godot in the command line. The same problem is still here. Here is the message I received in the terminal :

Vulkan devices:
  #0: AMD AMD Radeon Graphics (RADV RENOIR) - Supported, Integrated
  #1: NVIDIA NVIDIA GeForce RTX 3060 Laptop GPU - Supported, Discrete
  #2: Unknown llvmpipe (LLVM 15.0.7, 256 bits) - Supported, CPU
Vulkan API 1.3.260 - Forward+ - Using Vulkan Device #1: NVIDIA - NVIDIA GeForce RTX 3060 Laptop GPU
Optional extension VK_KHR_incremental_present not found
Optional extension VK_GOOGLE_display_timing not found
- Vulkan Variable Rate Shading supported:
  Pipeline fragment shading rate
  Primitive fragment shading rate
  Attachment fragment shading rate, min texel size: (16, 16), max texel size: (16, 16)
- Vulkan multiview supported:
  max view count: 32
  max instances: 134217727
- Vulkan subgroup:
  size: 32
  min size: 32
  max size: 32
  stages: STAGE_VERTEX, STAGE_TESSELLATION_CONTROL, STAGE_TESSELLATION_EVALUATION, STAGE_GEOMETRY, STAGE_FRAGMENT, STAGE_COMPUTE, STAGE_RAYGEN_KHR, STAGE_ANY_HIT_KHR, STAGE_CLOSEST_HIT_KHR, STAGE_MISS_KHR, STAGE_INTERSECTION_KHR, STAGE_CALLABLE_KHR, STAGE_TASK_NV, STAGE_MESH_NV
  supported ops: FEATURE_BASIC, FEATURE_VOTE, FEATURE_ARITHMETIC, FEATURE_BALLOT, FEATURE_SHUFFLE, FEATURE_SHUFFLE_RELATIVE, FEATURE_CLUSTERED, FEATURE_QUAD, FEATURE_PARTITIONED_NV

**Comment by clayjohn** [comment]
There's nothing obvious from that information. 

Can you test 4.3 dev 4 (https://godotengine.org/article/dev-snapshot-godot-4-3-dev-4/) and see if it works for you? We have done a few important fixes to problems that are similar to what you describe.

**Comment by ahmed-mo2nis** [comment]
> There's nothing obvious from that information.
> 
> Can you test 4.3 dev 4 (https://godotengine.org/article/dev-snapshot-godot-4-3-dev-4/) and see if it works for you? We have done a few important fixes to problems that are similar to what you describe.

I tried it to create a new project and everything seems to be working :) I created a new project with the Forward+ Renderer and changed it to Mobile to see if anything will crash or freeze. No problems so far. I created a simple scene as shown in these screenshots to fully test it. Thanks for fixing the issue :) 

![Screenshot from 2024-03-02 14-47-37](https://github.com/godotengine/godot/assets/40705255/e0e31694-be03-414b-8d18-53d7a9e4399b)
![Screenshot from 2024-03-02 14-48-01](https://github.com/godotengine/godot/assets/40705255/409887f7-08de-403e-8102-b825652e5f4f)
![Screenshot from 2024-03-02 14-48-40](https://github.com/godotengine/godot/assets/40705255/17796686-1bc2-4dc7-abb8-b0bfe13263d4)
![Screenshot from 2024-03-02 14-49-03](https://github.com/godotengine/godot/assets/40705255/bba37a6f-5aed-44a2-9020-945f9e18037a)
![Screenshot from 2024-03-02 14-49-31](https://github.com/godotengine/godot/assets/40705255/7696321c-cf41-4f16-b746-e1a975decbec)
![Screenshot from 2024-03-02 14-49-38](https://github.com/godotengine/godot/assets/40705255/df226bfe-8f46-448e-a76d-b78156d28e0b)

**Comment by clayjohn** [comment]
Amazing! Thank you for confirming that dev4 works for you. I will close this issue as resolved in 4.3.

For future reference I think https://github.com/godotengine/godot/pull/87340 is responsible for fixing this issue, but I am not 100% certain.
